Source/JavaScriptCore/jit/JITPropertyAccess.cpp

```diff
@@ -1809,7 +1809,7 @@ MacroAssemblerCodeRef<JITThunkPtrTag> JIT::generateOpGetFromScopeThunk(VM& vm)
             // Structure check covers var injection since we don't cache structures for anything but the GlobalObject. Additionally, resolve_scope handles checking for the var injection.
             jit.loadPtr(Address(metadataGPR, OpGetFromScope::Metadata::offsetOfStructure()), scratchGPR);
             slowCase.append(jit.branchTestPtr(Zero, scratchGPR));
-            jit.load32(Address(scratchGPR, Structure::structureIDOffset()), scratchGPR);
+            jit.emitEncodeStructure(scratchGPR, scratchGPR);
```

Comment:
> We don't need to do this now, but on platforms that aren't STRUCTURE_ID_WITH_SHIFT, we could just do a simple branch32 below, no need to actually encode.

Sorry, I don't understand that. Actually, I don't understand that piece of code, neither. Any explanation could be helpful.

Reply:
On address 64 platforms that don't use STRUCTURE_ID_WITH_SHIFT (e.g, macOS, Linux, but not iOS), the lower 32-bits of the Structure* pointer are the StructureID, so there is no need to do an extra move before comparing the bits. We could just do branch32 directly instead of emitEncodeStructureID followed by branch32.

On address 32 platforms, StructureID is Structure*, so the same applies there. 

It's not super important to do this, just wanted to point it out.